src/js/helpers.js

```diff
@@ -494,7 +494,7 @@ ${enterpriseSettings}`;
   }
   const debInstallationCode = `wget -q -O- https://get.mender.io/${
     isPreRelease && window.location.hostname.includes('staging') ? 'staging' : ''
-  } | sudo bash -s`;
+  } | sudo bash -s -- --demo`;
```

Comment:
This will only work for `https://get.mender.io/staging`, so merging it as-is will block any HM upgrade that we would like to do before Mender 2.7

I don't know if is the best solution, but in order to not add more hacks to the `gui` repo, my proposal is to modify production script to accept (but silently ignore) the new `--demo` flag.

Other ideas?

Reply:
preferably there would be yet another condition in there, as the `--demo` likely should only be applied during onboarding, but we know that user use that code snippet all the time - so we'll have to pass the onboarding state in there too and that function will be even more hackish... with that in mind (and the possible reduction in complexity): could we get the script to accept the flags directly ready for 2.7? 👍

and with regards to the staging limitation: would the additional flag actually disturb the `get.mender.io` script or would it just be ignored?

Reply:
> we know that user use that code snippet all the time

I wasn't aware it was used elsewhere. I thought it was only for onboarding. 

> with that in mind (and the possible reduction in complexity): could we get the script to accept the flags directly ready for 2.7?

I'm not sure I understand. Isn't this accepting them directly?

> This will only work for `https://get.mender.io/staging`, so merging it as-is will block any HM upgrade that we would like to do before Mender 2.7

Good thinking! This is easy to fix: Feature branch, and add an extra action point in https://tracker.mender.io/browse/MEN-4499.

Reply:
I am a bit confused, can we have chat about this today after SU?

Reply:
Discussed privately, will include a conditional in the UI for staging environment instead of using a feature branch. Will not add any other flags right now, since it's not part of this task which has already been going on for a long time.

Reply:
@mzedel: How about now?

Note that I left the original commit intact in the history, so that this is an easy revert later. I suppose this can be added as a cleanup item to https://tracker.mender.io/browse/MEN-4499.